src/Illuminate/Support/Siesta.php

```diff
@@ -0,0 +1,318 @@
+     * @return $this
+     */
+    public static function sleep($duration)
+    {
+        return static::for($duration)->seconds();
+    }
+
+    /**
+     * Sleep for the duration in minutes.
+     *
+     * @return $this
+     */
+    public function minutes()
+    {
+        $this->duration->addMinutes($this->pending);
+
+        $this->pending = 0;
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in minutes.
+     *
+     * @return $this
+     */
+    public function minute()
+    {
+        return $this->minutes();
+    }
+
+    /**
+     * Sleep for the duration in seconds.
+     */
+    public function seconds()
+    {
+        $this->duration->addSeconds($this->pending);
+
+        $this->pending = 0;
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in seconds.
+     *
+     * @return $this
+     */
+    public function second()
+    {
+        return $this->seconds();
+    }
+
+    /**
+     * Sleep for the duration in milliseconds.
+     *
+     * @return $this
+     */
+    public function milliseconds()
+    {
+        $this->duration->addMilliseconds($this->pending);
+
+        $this->pending = 0;
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in milliseconds.
+     *
+     * @return $this
+     */
+    public function millisecond()
+    {
+        return $this->milliseconds();
+    }
+
+    /**
+     * Sleep for the duration in microseconds.
+     *
+     * @return $this
+     */
+    public function microseconds()
+    {
+        $this->duration->addMicroseconds($this->pending);
+
+        $this->pending = 0;
+
+        return $this;
+    }
+
+    /**
+     * Sleep for the duration in microseconds.
+     *
+     * @return $this
+     */
+    public function microsecond()
+    {
+        return $this->microseconds();
+    }
+
+    /**
+     * Add additional time to sleep for.
+     *
+     * @param  int|float  $duration
+     * @return $this
+     */
+    public function and($duration)
+    {
+        $this->pending = $duration;
+
+        return $this;
+    }
+
+    /**
+     * Handle the object's destruction.
+     *
+     * @return void
+     */
+    public function __destruct()
+    {
+        if ($this->pending !== 0) {
+            throw new RuntimeException('Unknown Siesta duration unit.');
+        }
+
+        if ($this->duration->totalMicroseconds <= 0) {
+            $this->duration = CarbonInterval::seconds(0);
+        }
+
+        if (static::$fake) {
+            if ($this->capture) {
+                static::$sequence[] = $this;
+            }
+
+            return;
+        }
+
+        $remaining = $this->duration->copy();
+
+        if ((int) $remaining->totalSeconds > 0) {
+            sleep((int) $remaining->totalSeconds);
+
+            $remaining = $remaining->subSeconds((int) $remaining->totalSeconds);
+        }
+
+        if ($remaining->totalMicroseconds > 0) {
+            usleep($remaining->totalMicroseconds);
```

Comment:
Is this code correct? Seems that `totalMicroseconds` may be a `float`, while `usleep` only works with `int`.

Reply:
I don't believe this is actually an issue. I'm unable to make this a float, so I can't write a test for it.

I will cast them to an int, just to be safe, but not sure it'll ever be a float tbh.